<launch>

  <group ns="rocket">
    <rosparam file="$(find tvc_simulator)/config/prototypes/drone/rocket_parameters.yaml" />
  </group>

  <group ns="environment">
    <rosparam file="$(find tvc_simulator)/config/environment_parameters.yaml" />
  </group>

  <group ns="perturbation">
    <rosparam file="$(find tvc_simulator)/config/perturbations_parameters.yaml" />
  </group>

  <group ns="visualization">
    <rosparam file="$(find tvc_simulator)/config/prototypes/drone/visualization_parameters.yaml" />
  </group>
''

  <node name="control" pkg="tvc_simulator" type="drone_control" output ="screen"/>
  <node name="guidance" pkg="tvc_simulator" type="guidance" output ="screen"  />
  <node name="integrator" pkg="tvc_simulator" type="integrator.py" cwd = "node" output ="screen"/>
  <node name="disturbance" pkg="tvc_simulator" type="disturbance.py" cwd = "node"/>
  <node name="data_fusion" pkg="tvc_simulator" type="data_fusion" cwd = "node"/>

  <node name="visualization" pkg="tvc_simulator" type="visualization" />
  <node name="rocket_state" pkg="rostopic" type="rostopic" args="echo /rocket_state"  /> 
  <node name="rqt_gui" pkg="rqt_gui" type="rqt_gui" args="--perspective-file $(find tvc_simulator)/rviz/drone_GUI.perspective"/>
<!--  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find tvc_simulator)/rviz/rocket_config.rviz" />-->

</launch>


